storage/replica.go

```diff
@@ -1559,6 +1559,10 @@ func (r *Replica) addWriteCmd(
 				} else {
 					log.Warningf(ctx, "unable to cancel expired Raft command %s", ba)
 				}
+			case <-r.store.stopper.ShouldQuiesce():
```

Comment:
I'm extracting this into a separate PR since we're seeing more failures related to this.